roles/deploy-aws-infra/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    tags:
+      Name: "{{ env_id }}-route-table-gw"
+      env_id: "{{ env_id }}"
+    subnets:
+      - "{{ new_subnets.subnet.id }}"
+    routes:
+      - dest: "0.0.0.0/0"
+        gateway_id: "{{ new_gw.gateway_id }}"
+
+# Create local facts with new subnet id
+- name: Register new created objects as facts
+  set_fact:
+    aws_subnet: "{{ new_subnets.subnet.id }}"
+
+- name: Create Master and etcd instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: m4.xlarge
+    group: "{{ aws_master_sgroups }}"
+    termination_protection: yes
+    key_name: "{{ aws_key_name }}"
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    region: "{{ aws_region }}"
+    vpc_subnet_id: "{{ aws_subnet }}"
+    exact_count: 1
+    count_tag:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+    instance_tags:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+      user: sre
+      role: ocp
+      group: masters_aws
+      node_labels: '{"region": "default"}'
+      ocp_role: "{{ master_tag }} {{ etcd_tag}}"
+      env_id: "{{ env_id }}"
+  with_sequence:
+    count="{{ aws_num_masters }}"
+  register: instances_master_etcd
+
+- name: Create Infra instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: i3.xlarge
+    group: "{{ aws_infra_sgroups }}"
+    termination_protection: yes
+    key_name: "{{ aws_key_name }}"
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    region: "{{ aws_region }}"
+    vpc_subnet_id: "{{ aws_subnet }}"
+    exact_count: 1
+    count_tag:
+      Name: "{{ env_id }}-ocp-infra-{{ item }}"
+    instance_tags:
+      Name: "{{ env_id }}-ocp-infra-{{ item }}"
+      user: sre
+      role: ocp
+      group: infra_aws
+      node_labels: '{"region": "infra"}'
+      ocp_role: "{{ infranode_tag }}"
+      env_id: "{{ env_id }}"
+  with_sequence:
+    count="{{ aws_num_infra }}"
+  register: instances_infra
+
+- name: Create Compute instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: m4.xlarge
+    group: "{{ aws_node_sgroups }}"
+    termination_protection: yes
+    key_name: "{{ aws_key_name }}"
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    region: "{{ aws_region }}"
+    vpc_subnet_id: "{{ aws_subnet }}"
+    exact_count: 1
+    count_tag:
+      Name: "{{ env_id }}-ocp-node-{{ item }}"
+    instance_tags:
+      Name: "{{ env_id }}-ocp-node-{{ item }}"
+      user: sre
+      role: ocp
+      group: nodes_aws
+      node_labels: '{"region": "primary"}'
+      ocp_role: "{{ computenode_tag }}"
+      env_id: "{{ env_id }}"
+  with_sequence:
+    count="{{ aws_num_nodes }}"
+  register: instances_compute
+
+- name: Give some time until the instances are running before attaching storage
+  pause:
+    seconds: 30
```

Comment:
Why 30? Is there a check we can do instead of a fixed sleep?

Reply:
There are 2 options here:
- Create a loop and check every N seconds until a limit of M
- Use 'wait' option for ec2 module so the task will not continue until the instances are running. This one has an inconvenient: instances are created one by one instead all at the same time

Reply:
Ok, yeah, it just seems that 30 seconds may be a potential failure on a day with a slow connection. There are various ways to do the "wait-loop" to check and break out of the loop if a positive response. Maybe consider something like that.

Reply:
@oybed I've finally used the 'wait' option as from the tests I've done it looks to be safer and more reliable.

Reply:
ok - hopefully that doesn't all too much wait time to the execution ... we can do a follow-up though to enhance it later on.